funding/manager.go

```diff
@@ -3601,3 +3659,144 @@ func (f *Manager) deleteChannelOpeningState(chanPoint *wire.OutPoint) error {
 		return bucket.Delete(outpointBytes.Bytes())
 	}, func() {})
 }
+
+// Here we wait for confirmation to update the real short channel id for this
+// not yet confirmed channel.
+func (f *Manager) updateShortChannelID(ch *channeldb.OpenChannel) (
+	*lnwire.ShortChannelID, error) {
+
+	log.Infof("waiting for confirmation to update short channel id")
+	confirmedChan := ch
+	confChan := make(chan *confirmedChannel, 1)
+	cancelChan := make(chan struct{})
+	skipChan := make(chan lnwire.ShortChannelID)
+
+	f.wg.Add(1)
+	go f.waitForFundingConfirmation(ch, cancelChan, confChan, 1)
+	defer close(skipChan)
+	defer close(cancelChan)
+
+	select {
+	case conf, ok := <-confChan:
+		var err error
+		if !ok {
+			return nil, fmt.Errorf("failed to wait for funding confirmation")
+		}
+		confirmedChan, err = f.cfg.FindChannel(lnwire.NewChanIDFromOutPoint(&ch.FundingOutpoint))
+		if err != nil {
+			return nil, fmt.Errorf("failed to find channel %v %v", ch.ShortChannelID, err)
+		}
+		err = f.cfg.Wallet.ValidateChannel(confirmedChan, conf.fundingTx)
+		if err != nil {
+			return nil, fmt.Errorf("unable to validate channel: %v", err)
+		}
+		if err := confirmedChan.MarkAsOpen(conf.shortChanID); err != nil {
+			return nil, fmt.Errorf("error setting channel pending flag to "+
+				"false: %v", err)
+		}
+		if err := f.cfg.ReportShortChanID(confirmedChan.FundingOutpoint); err != nil {
+			log.Errorf("unable to report short chan id: %v", err)
+		}
+	case <-f.quit:
+		// The fundingManager is shutting down, and will resume wait on
+		// startup.
+		return nil, ErrFundingManagerShuttingDown
+	}
+	return &confirmedChan.ShortChannelID, nil
+}
+
+type fakeIDsManager struct {
+	db *channeldb.DB
+}
+
+func newFakeIDsManager(channels []*channeldb.OpenChannel,
```

Comment:
Can likely extract this into a new file to isolate the logic a bit more.

Reply:
Why do we need to prune things all at once, rather than having the funding state machine for each channel prune them out once things are confirmed?

Reply:
> Can likely extract this into a new file to isolate the logic a bit more.

Sounds like a good idea, will do that.

Reply:
> Why do we need to prune things all at once, rather than having the funding state machine for each channel prune them out once things are confirmed?

We actually prune them as part of the funding workflow but I added this prune logic to cover some edge cases that to remember them I need to refresh my memory.  I will address this once my memory comes back to me.